frontend/src/metabase/query_builder/components/CompareAggregations/utils.ts

```diff
@@ -91,6 +137,71 @@ export const canSubmit = (
   return isPeriodValid && areColumnsValid;
 };
 
+type UpdatedQuery = {
+  query: Lib.Query;
+  addedAggregations: Lib.ExpressionClause[];
+};
+
+export function updateQueryWithCompareOffsetAggregations(
+  query: Lib.Query,
+  stageIndex: number,
+  aggregation: Lib.AggregationClause | Lib.ExpressionClause,
+  offset: "" | number,
+  columns: ColumnType[],
+  columnAndBucket: BreakoutColumnAndBucket,
+): UpdatedQuery | null {
+  if (!aggregation || offset === "" || !columnAndBucket) {
+    return null;
+  }
+
+  let nextQuery = query;
+  const column = Lib.withTemporalBucket(
+    columnAndBucket.column,
+    columnAndBucket.bucket,
+  );
+
+  let { breakoutIndex } = columnAndBucket;
+  if (breakoutIndex !== null) {
+    // replace the breakout
+    const breakout = Lib.breakouts(nextQuery, stageIndex)[breakoutIndex];
+    nextQuery = Lib.replaceClause(nextQuery, stageIndex, breakout, column);
+  } else {
+    // add the breakout
+    nextQuery = Lib.breakout(nextQuery, stageIndex, column);
+    breakoutIndex = Lib.breakouts(nextQuery, stageIndex).length - 1;
+  }
+
+  if (breakoutIndex !== 0) {
```

Comment:
nit: it won't make a difference now but we can future-proof this a bit:

```suggestion
  if (breakoutIndex > 0) {
```